pennylane/templates/subroutines/select.py

```diff
@@ -70,44 +80,241 @@ class Select(Operation):
+
+        .. code-block::
+
+            0: ─╭●──       ─●─╮─
+            1: ─├●──  and  ─●─┤─
+            2:  ╰───       ───╯
+
+        for :class:`~.TemporaryAND` and its adjoint, respectively, and skipped drawing the
+        auxiliary wires in areas where they are guaranteed to be in the state :math:`|0\rangle`.
+        We will need three simplification rules for pairs of ``TemporaryAND`` gates:
+
+        .. code-block::
+
+            ─○─╮─╭○──   ──     ─○─╮─╭○──   ─╭○─       ─○─╮─╭●──   ─╭●────
+            ─○─┤─├○── = ──,    ─○─┤─├●── = ─│──, and  ─●─┤─├○── = ─│──╭●─.
+            ───╯ ╰───   ──     ───╯ ╰───   ─╰X─       ───╯ ╰───   ─╰X─╰X─
+
+        Applying these simplifications reduces the computational cost of the ``Select``
+        template:
+
+        .. code-block::
+
+            0:    ─╭○────────────────╭○──────────────────╭●─────────────────────╭●─────────────────●╮─┤
+            1:    ─├○────────────────│───────────────────│──╭●──────────────────│──────────────────●┤─┤
+            aux0:  ╰─╭●─────╭●────●╮─╰X─╭●─────╭●─────●╮─╰X─╰X─╭●─────╭●─────●╮─╰X─╭●─────╭●─────●╮─╯ │
+            2:    ───├○─────│─────●┤────├○─────│──────●┤───────├○─────│──────●┤────├○─────│──────●┤───┤
+            aux1:    ╰─╭●───╰X─╭●──╯    ╰─╭●───╰X──╭●──╯       ╰─╭●───╰X──╭●──╯    ╰─╭●───╰X──╭●──╯   │
+            3:    ─────├U(M0)──├U(M1)─────├U(M2)───├U(M3)────────├U(M4)───├U(M5)─────├U(M6)───├U(M7)──┤
+            4:    ─────╰U(M0)──╰U(M1)─────╰U(M2)───╰U(M3)────────╰U(M4)───╰U(M5)─────╰U(M6)───╰U(M7)──┤
+
+        An additional cost reduction then results from the fact that ``TemporaryAND`` gates
+        (``adjoint(TemporaryAND)`` gates) take four (zero) :class:`~T` gates, instead of the seven
+        ``T`` gates required by a decomposition of :class:`~Toffoli`.
+
+        For general :math:`c` and :math:`K=2^c`, the decomposition takes a similar form, with
+        alternating control and auxiliary wires.
+
+        An implementation of this decomposition is easily achieved in the following steps:
+        We first define a recursive subroutine ``R`` that will be used in the main function:
+        we are given :math:`L` operators and :math:`2 \lceil\log_2(L)\rceil + 1` control and
+        auxiliary wires. If ``L=0``, ``R`` does not apply any operators.
+        If ``L=1``, the single operator is applied, controlled on the first control wire.
+
+        In all other cases, ``R`` applies the circuit
+
+        .. code-block::
+
+            aux_j:   ─╭●────╭●────●─╮─
+            j+1:     ─├○────│─────●─┤─
+            aux_j+1:  ╰──■──╰X─■────╯
+
+        where each box symbolizes a call to ``R`` itself, on the next recursion level.
+        These next-level calls use
+        :math:`L' = 2^{\lceil\log_2(L)\rceil-1}` (i.e. half of :math:`L`, rounded up to the next
+        power of two) and :math:`L-L'` (i.e. the rest) operators, respectively.
+
+        With ``R`` defined, we are ready to outline the main function:
```

Comment:
is this a general procedure? Like how can it always be separated in four quarters? (is it even?)